app.rb

```diff
@@ -0,0 +1,180 @@
+    if @people.empty?
+      puts 'There are no people'
+    else
+      @people.each do |person|
+        puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+      display
+    end
+    display
+  end
+
+  def create_book
+    print 'Title of the book: '
+    title = gets.chomp
+    print 'Author of the book: '
+    author = gets.chomp
+    book = Book.new(title, author)
+    @books << book
+    puts 'Book created successfully'
+    display
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    input = gets.chomp
+    case input
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Please choose a valid option'
+    end
+  end
+
+  def create_student
+    print 'Add name: '
+    name = gets.chomp
+    print 'Add age: '
+    age = gets.chomp
+    print 'Add a classroom: '
+    classroom = gets.chomp
+    print 'Add parent permission (True or False)'
+    parent_permission = gets.chomp
+    permission = case parent_permission.downcase
+                 when 'true'
+                   true
+                 when 'false'
+                   false
+                 end
+    student = Student.new(classroom, age, name, permission)
+    @people << student
+    puts 'Student created successfully'
+    display
+  end
+
+  def create_teacher
+    print 'Add name: '
+    name = gets.chomp
+    print 'Add age: '
+    age = gets.chomp
+    print 'Add a specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(specialization, age, name)
+    @people << teacher
+    puts 'Teacher created successfully'
+    display
+  end
+
+  def create_rental
+    if @books.length.positive?
+      puts 'Select a book that by there number'
+      @books.each_with_index do |book, index|
+        puts "[#{index}] Title: #{book.title}, Author: #{book.author}"
+      end
+      books = gets.chomp.capitalize
+      puts 'Select a person to rent the book by there number'
+      @people.each_with_index do |person, index|
+        puts "[#{index}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+      persons = gets.chomp.capitalize
+      puts 'Add date of rental [yyyy-mm-dd]]'
+      date = gets.chomp
+      rental = Rental.new(date, @books[books.to_i], @people[persons.to_i])
+      @rentals << rental
+      puts 'Rental created successfully'
+      display
+    else
+      display
+      puts 'There are no books'
+    end
+  end
+end
+
+def list_rentals
+  @people.each do |person|
+    puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+  end
+  puts 'Select a person to see their rentals by there id'
+  rental_id = gets.chomp.to_i
+  @rentals.each do |rental|
+    if rental.person.id == rental_id
+      puts "Date: #{rental.date}, Book: #{rental.book.title} by #{rental.book.author}, Person: #{rental.person.name}"
+    end
+    display
+  end
+end
```

Comment:
While listing the rentals, it only displays one rental for student even though I did create multiple and no rentals for teacher. Kindly fix this issue. I have recorded [this video](https://www.loom.com/share/4351345e7baf4807af273d62bfab9994) where i demonstrated what exactly is the issue.